crates/store/re_sorbet/src/chunk_batch.rs

```diff
@@ -1,26 +1,194 @@
-use arrow::array::RecordBatch as ArrowRecordBatch;
-use arrow::datatypes::Schema as ArrowSchema;
+use std::sync::Arc;
 
+use arrow::{
+    array::{
+        Array as _, ArrayRef as ArrowArrayRef, AsArray, RecordBatch as ArrowRecordBatch,
+        RecordBatchOptions, StructArray as ArrowStructArray,
+    },
+    datatypes::{
+        DataType as ArrowDataType, Field as ArrowField, FieldRef as ArrowFieldRef,
+        Fields as ArrowFields, Schema as ArrowSchema,
+    },
+};
+
+use re_arrow_util::{arrow_util::into_arrow_ref, ArrowArrayDowncastRef};
 use re_log_types::EntityPath;
+use re_types_core::ChunkId;
+
+use crate::{
+    chunk_schema::InvalidChunkSchema, ArrowBatchMetadata, ChunkSchema, ComponentColumnDescriptor,
+    RowIdColumnDescriptor, TimeColumnDescriptor, WrongDatatypeError,
+};
 
 #[derive(thiserror::Error, Debug)]
-pub enum ChunkBatchError {
-    #[error("Missing record batch metadata key `{0}`")]
-    MissingRecordBatchMetadataKey(&'static str),
+pub enum MismatchedChunkSchemaError {
+    #[error("{0}")]
+    Custom(String),
+
+    #[error(transparent)]
+    WrongDatatypeError(#[from] WrongDatatypeError),
+}
+
+impl MismatchedChunkSchemaError {
+    pub fn custom(s: impl Into<String>) -> Self {
+        Self::Custom(s.into())
+    }
 }
 
-/// The arrow [`ArrowRecordBatch`] representation of a Rerun chunk.
+/// The [`ArrowRecordBatch`] representation of a Rerun chunk.
```

Comment:
I'm getting super confused regarding terminology here: so this is _not_ a Sorbet chunk, this is a Rerun chunk (i.e. single-entity kinda thing), right? Are Sorbet chunks defined somewhere or have we just dropped them for now?